// actions-on-google/dist/index.js
"use strict";
/**
 * Copyright 2018 Google Inc. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.smarthome = exports.actionssdk = exports.ActionsSdkConversation = exports.dialogflow = exports.DialogflowConversation = exports.UnauthorizedError = exports.SoloHelper = exports.Helper = exports.DigitalPurchaseCheck = exports.CompletePurchase = exports.DeliveryAddress = exports.TransactionRequirements = exports.TransactionDecision = exports.DeepLink = exports.Place = exports.RegisterUpdate = exports.UpdatePermission = exports.SignIn = exports.Permission = exports.NewSurface = exports.List = exports.DateTime = exports.Confirmation = exports.Carousel = exports.Suggestions = exports.RichResponse = exports.HtmlResponse = exports.OrderUpdate = exports.MediaResponse = exports.MediaObject = exports.BrowseCarousel = exports.BrowseCarouselItem = exports.Table = exports.BasicCard = exports.Button = exports.OpenUrlAction = exports.Image = exports.SimpleResponse = exports.LinkOutSuggestion = exports.Conversation = void 0;
var actionssdk_1 = require("./service/actionssdk");
Object.defineProperty(exports, "Conversation", { enumerable: true, get: function () { return actionssdk_1.Conversation; } });
Object.defineProperty(exports, "LinkOutSuggestion", { enumerable: true, get: function () { return actionssdk_1.LinkOutSuggestion; } });
Object.defineProperty(exports, "SimpleResponse", { enumerable: true, get: function () { return actionssdk_1.SimpleResponse; } });
Object.defineProperty(exports, "Image", { enumerable: true, get: function () { return actionssdk_1.Image; } });
Object.defineProperty(exports, "OpenUrlAction", { enumerable: true, get: function () { return actionssdk_1.OpenUrlAction; } });
Object.defineProperty(exports, "Button", { enumerable: true, get: function () { return actionssdk_1.Button; } });
Object.defineProperty(exports, "BasicCard", { enumerable: true, get: function () { return actionssdk_1.BasicCard; } });
Object.defineProperty(exports, "Table", { enumerable: true, get: function () { return actionssdk_1.Table; } });
Object.defineProperty(exports, "BrowseCarouselItem", { enumerable: true, get: function () { return actionssdk_1.BrowseCarouselItem; } });
Object.defineProperty(exports, "BrowseCarousel", { enumerable: true, get: function () { return actionssdk_1.BrowseCarousel; } });
Object.defineProperty(exports, "MediaObject", { enumerable: true, get: function () { return actionssdk_1.MediaObject; } });
Object.defineProperty(exports, "MediaResponse", { enumerable: true, get: function () { return actionssdk_1.MediaResponse; } });
Object.defineProperty(exports, "OrderUpdate", { enumerable: true, get: function () { return actionssdk_1.OrderUpdate; } });
Object.defineProperty(exports, "HtmlResponse", { enumerable: true, get: function () { return actionssdk_1.HtmlResponse; } });
Object.defineProperty(exports, "RichResponse", { enumerable: true, get: function () { return actionssdk_1.RichResponse; } });
Object.defineProperty(exports, "Suggestions", { enumerable: true, get: function () { return actionssdk_1.Suggestions; } });
var actionssdk_2 = require("./service/actionssdk");
Object.defineProperty(exports, "Carousel", { enumerable: true, get: function () { return actionssdk_2.Carousel; } });
Object.defineProperty(exports, "Confirmation", { enumerable: true, get: function () { return actionssdk_2.Confirmation; } });
Object.defineProperty(exports, "DateTime", { enumerable: true, get: function () { return actionssdk_2.DateTime; } });
Object.defineProperty(exports, "List", { enumerable: true, get: function () { return actionssdk_2.List; } });
Object.defineProperty(exports, "NewSurface", { enumerable: true, get: function () { return actionssdk_2.NewSurface; } });
Object.defineProperty(exports, "Permission", { enumerable: true, get: function () { return actionssdk_2.Permission; } });
Object.defineProperty(exports, "SignIn", { enumerable: true, get: function () { return actionssdk_2.SignIn; } });
Object.defineProperty(exports, "UpdatePermission", { enumerable: true, get: function () { return actionssdk_2.UpdatePermission; } });
Object.defineProperty(exports, "RegisterUpdate", { enumerable: true, get: function () { return actionssdk_2.RegisterUpdate; } });
Object.defineProperty(exports, "Place", { enumerable: true, get: function () { return actionssdk_2.Place; } });
Object.defineProperty(exports, "DeepLink", { enumerable: true, get: function () { return actionssdk_2.DeepLink; } });
Object.defineProperty(exports, "TransactionDecision", { enumerable: true, get: function () { return actionssdk_2.TransactionDecision; } });
Object.defineProperty(exports, "TransactionRequirements", { enumerable: true, get: function () { return actionssdk_2.TransactionRequirements; } });
Object.defineProperty(exports, "DeliveryAddress", { enumerable: true, get: function () { return actionssdk_2.DeliveryAddress; } });
Object.defineProperty(exports, "CompletePurchase", { enumerable: true, get: function () { return actionssdk_2.CompletePurchase; } });
Object.defineProperty(exports, "DigitalPurchaseCheck", { enumerable: true, get: function () { return actionssdk_2.DigitalPurchaseCheck; } });
Object.defineProperty(exports, "Helper", { enumerable: true, get: function () { return actionssdk_2.Helper; } });
Object.defineProperty(exports, "SoloHelper", { enumerable: true, get: function () { return actionssdk_2.SoloHelper; } });
Object.defineProperty(exports, "UnauthorizedError", { enumerable: true, get: function () { return actionssdk_2.UnauthorizedError; } });
var dialogflow_1 = require("./service/dialogflow");
Object.defineProperty(exports, "DialogflowConversation", { enumerable: true, get: function () { return dialogflow_1.DialogflowConversation; } });
Object.defineProperty(exports, "dialogflow", { enumerable: true, get: function () { return dialogflow_1.dialogflow; } });
__exportStar(require("./service/dialogflow/api/v1"), exports);
__exportStar(require("./service/dialogflow/api/v2"), exports);
var actionssdk_3 = require("./service/actionssdk");
Object.defineProperty(exports, "ActionsSdkConversation", { enumerable: true, get: function () { return actionssdk_3.ActionsSdkConversation; } });
Object.defineProperty(exports, "actionssdk", { enumerable: true, get: function () { return actionssdk_3.actionssdk; } });
__exportStar(require("./service/actionssdk/api/v2"), exports);
var smarthome_1 = require("./service/smarthome");
Object.defineProperty(exports, "smarthome", { enumerable: true, get: function () { return smarthome_1.smarthome; } });
__exportStar(require("./service/smarthome/api/v1"), exports);
